pkg/etcdcli/etcdcli.go

```diff
@@ -255,33 +255,16 @@ func (g *etcdClientGetter) MemberUpdatePeerURL(ctx context.Context, id uint64, p
 	return err
 }
 
-func (g *etcdClientGetter) MemberRemove(ctx context.Context, member string) error {
-	g.eventRecorder.Eventf("MemberRemove", "removing member %q", member)
-
+func (g *etcdClientGetter) MemberRemove(ctx context.Context, memberID uint64) error {
 	cli, err := g.getEtcdClient()
 	if err != nil {
 		return err
 	}
-	membersResp, err := cli.MemberList(ctx)
-	if err != nil {
-		return nil
-	}
-
-	for _, m := range membersResp.Members {
-		if m.Name == member {
-			cctx, ccancel := context.WithCancel(ctx)
-			defer ccancel()
-
-			_, err = cli.MemberRemove(cctx, m.ID)
-			if err != nil {
-				return err
-			}
-			return nil
-		}
+	_, err = cli.MemberRemove(ctx, memberID)
+	if err == nil {
+		g.eventRecorder.Eventf("MemberRemove", "removed member with ID: %v", memberID)
```

Comment:
I think at a glance it's more useful to see the member name (along with the ID for unstarted members) in the event, so you can more easily tie it to the node that was removed. It's harder to discern that with just the member ID alone.
This is what I was trying to preserve by passing in the entire member struct.
https://github.com/hasbro17/cluster-etcd-operator/blob/62ce76b24d48c749c16f0503111bd4752149a69b/pkg/etcdcli/etcdcli.go#L344

But I'm okay with doing a follow up for this later.